docs/src/architecture/notes.md

```diff
@@ -1,73 +1,194 @@
+ In addition, a note has **metadata** including the sender, tag, and number of assets.
 
-`hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)`
+# Note's lifecycle
+New notes are being created when executing a transaction. After verifying the transaction proof the Operator adds either only the note hash or the full note data to the [Notes DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database). Notes can be produced and consumed locally by users in local transactions or by the operator in a network transaction. Note consumption requires the transacting party to know the note data to compute the nullifier. After successful verification, the Operator sets the corresponding entry in the Nullifier DB to `1`. 
 
-This achieves the following properties:
-- Every note can be reduced to a single unique hash.
-- To compute a note's hash, we do not need to know the note's `serial_num`. Knowing the hash
-    of the `serial_num` (as well as `script_hash`, `input_hash` and `note_vault`) is sufficient.
-- Moreover, the note hash can be computed having the [recipient](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html#note-recipient) and note vault, as `hash(recipient, vault_hash)`.
-- We compute the hash of `serial_num` as `hash(serial_num, [0; 4])` to simplify processing within the VM.
-
-# Note nullifier
-The nullifier is the note's index in the [Nullifier DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#nullifier-database). The Nullifier DB stores the information whether a note was already consumed.
+<p align="center">
+    <img src="../diagrams/architecture/note/Note_life_cycle.png">
+</p>
 
-The nullifier is computed as `hash(serial_num, script_hash, input_hash, vault_hash)`.
+The following section will explain, how notes are created, stored, discovered and consumed. 
+
+## Note creation
+Notes are created in Miden transactions. If a valid Miden transaction outputting an `OutputNote` gets verified by the Miden Operator it gets added to the Notes DB.
+
+### Creating the note object 
+Most users will create new note objects using the Miden Client (ToDo define how to). It is also possible to create note objects directly in [Rust](https://github.com/0xPolygonMiden/miden-base/blob/main/miden-lib/src/notes/mod.rs) or by using a [transaction kernel procedure](https://github.com/0xPolygonMiden/miden-base/blob/e4cba6a5d9581f76604cd89bd05d129b3c84e254/miden-lib/asm/kernels/transaction/api.masm#L421) in MASM. 
+
+
+## The note script 
+Every note has a script which gets executed at note consumption. The script allows for more than just the transferring of assets. It is always executed in the context of a single account, and thus, may invoke zero or more of the [account's functions](https://0xpolygonmiden.github.io/miden-base/architecture/accounts.html#code). Note scripts can become arbirarly complex due to the underlying Turing-complete Miden VM. 
```

Comment:
Excellent questions.

1. I think atm we need to have a basic script at least: https://github.com/0xPolygonMiden/miden-base/blob/main/objects/src/notes/script.rs#L22

2. I changed it, is it better?

3. Good point, but this will stay true for at least six month

4. Nice rephrasing, I used your suggestion 

```
Every note has a script which gets executed at note consumption. It is always executed in the context of a single account, and thus, may invoke zero or more of the [account's functions](https://0xpolygonmiden.github.io/miden-base/architecture/accounts.html#code). The script allows for more than just the transferring of assets, they could be of arbitrary complexity thanks to the Turing completeness of the Miden VM 
```

Reply:
let me know what you think.